plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/OpenShiftJobs.java

```diff
@@ -39,6 +44,15 @@ private OpenShiftJobs() {}
 	 * Creates a {@link DeleteResourceJob} to delete an OpenShift {@link IProject}.
 	 */
 	public static DeleteResourceJob createDeleteProjectJob(final IProject project) {
+		return createDeleteProjectJob(project, null);
+	}
+	
+	/**
+	 * Creates a {@link DeleteResourceJob} to delete an OpenShift {@link IProject}.
+	 * @param project
+	 * @param callIfNotOK - A callback to execite if the job does not return OK status 
+	 */
+	public static DeleteResourceJob createDeleteProjectJob(final IProject project, Runnable callIfNotOK) {
```

Comment:
the Runnable "callIfNotOK" looks abit redundant/duplicate to what IJobChangeListener offers: The one that creates the delete job could attach a job change listener to it and implement #done, inspecting the IStatus that's in the IJobChangeEvent he'll get.